src/main/java/util/ApiQuotaManager.java

```diff
@@ -0,0 +1,169 @@
+package util;
+
+import backend.Logic;
+import backend.github.ApiQuotaInfo;
+import org.apache.logging.log4j.Logger;
+import ui.UI;
+import util.events.RefreshTimerTriggeredEvent;
+import util.events.RefreshTimerTriggeredEventHandler;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Manages the API quota usage and periodic refreshes of the data store.
+ */
+public class ApiQuotaManager {
+
+    private static final Logger logger = HTLog.get(ApiQuotaManager.class);
+
+    /**
+     * The amount that is set aside for manual operations by the user.
+     */
+    private static final int API_QUOTA_BUFFER = 200;
+
+    /**
+     * The default refresh period.
+     */
+    private static final int DEFAULT_REFRESH_PERIOD_IN_MINS = 1;
+
+    /**
+     * The previous amount of the available remaining API requests.
+     */
+    private int previousRemainingApiRequests = 0;
```

Comment:
The usage of this variable happens before it gets updated. So It is actually using the previous remaining Api Request. So i thought it will be more understandable if i put the prefix previous?

`int difference = previousRemainingApiRequests - remainingRequests;`

If changed it to 

`int difference = remainingApiRequests - remainingRequests;`

It looks abit confusing?